api/Dockerfile

```diff
@@ -14,8 +14,16 @@ RUN npm install -g npm
 ARG CONFIG_ID
 RUN echo "CONFIG_ID=$CONFIG_ID"
 
-# Install all the packages for the workspaces
-RUN npm install --only=prod
+# Install all packages necessary for compilation, then remove the devDependencies
+RUN npm install && npm run build-config && npm run build-compile && npm prune --production
```

Comment:
The reason why we need to install all dependencies and then remove some is because `typescript` and the `@types/*` packages are `devDependencies` necessary for compilation. Since they're not necessary for running the app, we can remove them after compilation